src/derp.c

```diff
@@ -198,35 +227,35 @@ static void derp_transmit_task(void *arg) {
 
 	// Prepare TLS configuration
 	esp_tls_cfg_t tls_cfg = {0};
-	tls_cfg.cacert_buf = cacert;
+	tls_cfg.cacert_buf = (unsigned char*) cacert;
 	tls_cfg.cacert_bytes = strlen(cacert) + 1;
 	tls_cfg.skip_common_name = true;
 
 	// Try to establish TLS connection
-	const char *derp_ip = "149.88.19.146"; // TODO: make configurable
-	err = esp_tls_conn_new_sync(derp_ip, strlen(derp_ip), 8765, &tls_cfg, dev->derp.tls);
+	err = esp_tls_conn_new_sync(dev->derp.endpoint.addr, dev->derp.endpoint.addr_len, dev->derp.endpoint.port, &tls_cfg, dev->derp.tls);
 	if (err == -1) {
 		ESP_LOGE(TAG, "Failed to attempt TLS connection address %d", err);
 		goto ret;
 	}
-	ESP_LOGE(TAG, "Connection Established!");
+	ESP_LOGI(TAG, "Connection Established!");
 
 	dev->derp.conn_state = CONN_STATE_TCP_CONNECTING;
-
+	char http_req[128];
 	// Prepare HTTP upgrade request
-	const char * http_req =
-		"GET /derp HTTP/1.1\r\n"
-		"Host: 149.88.19.146\r\n"
-		"Connection: Upgrade\r\n"
-		"Upgrade: WebSocket\r\n"
-		"User-Agent: esp32/v1.0.0 esp\r\n\r\n";
+    snprintf(http_req, 128,
+        "GET /derp HTTP/1.1\r\n"
+	    "Host: %s\r\n"
+        "Connection: Upgrade\r\n"
+        "Upgrade: WebSocket\r\n"
+        "User-Agent: esp32/v1.0.0 esp\r\n\r\n",
```

Comment:
In libtelio, user agent is set to `telio/{OS}`, so I guess we should stick here with `telio/esp32`